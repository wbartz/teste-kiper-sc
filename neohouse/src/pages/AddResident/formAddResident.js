import PropTypes from 'prop-types';
import React, { useState } from 'react';
import { useForm } from 'react-hook-form';
import Form from './form';
import validationSchema from './validations';
import Success from './success';
import './index.scss';

const FormResident = ({ history, location, addResident }) => {
  const [showSuccess, setShowSuccess] = useState(false);
  const [user, setUser] = useState({});
  const { handleSubmit, control, errors } = useForm({
    validationSchema,
  });
  const { apartment_id } = location.state;

  const onSuccess = (values) => {
    setShowSuccess(true);
    setUser(values);
  };

  const handleAdd = (values) => {
    addResident({ ...values, apartment_id }, onSuccess);
  };

  return (
    <div className="page add-resident-page">
      <form onSubmit={handleSubmit(handleAdd)}>
        <Form user={user} errors={errors} control={control} />
        <div className="row buttons">
          <div className="col m8">
            <div className="col m2 lef">
              <button
                type="button"
                className="btn btn-flat"
                disabled={errors.length}
                onClick={() => history.goBack()}
              >
                Cancelar
              </button>
            </div>
            <div className="col m2 right">
              <button
                type="submit"
                className="btn btn-submit primary"
                disabled={errors.length}
              >
                Salvar
              </button>
            </div>
          </div>
        </div>
      </form>

      <Success
        show={showSuccess}
        user={user}
        onClose={() => {
          setShowSuccess(false);
        }}
      />
    </div>
  );
};

FormResident.propTypes = {
  history: PropTypes.shape({ push: PropTypes.func }).isRequired,
  location: PropTypes.oneOfType([PropTypes.object, PropTypes.array]).isRequired,
  addResident: PropTypes.func.isRequired,
};

export default FormResident;
