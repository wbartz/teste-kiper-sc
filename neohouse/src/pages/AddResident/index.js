import React, { useContext } from 'react';
import { withRouter } from 'react-router-dom';
import { AppContext } from '../../containers/StoreProvider';
import FormAddResident from './formAddResident';

export default withRouter((props) => (
  <FormAddResident {...useContext(AppContext)} {...props} />
));
