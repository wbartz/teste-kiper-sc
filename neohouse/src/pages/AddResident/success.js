import PropTypes from 'prop-types';
import React from 'react';
import { withRouter } from 'react-router';
import Modal from '../../components/Modal';

export const Success = ({ show, user, onClose }) => {
  const handleFinish = () => onClose();

  return (
    <Modal className="edit-modal" show={show} title="Pronto!">
      <div className="row">
        <span>
          O morador
          <b> {user?.full_ame} </b>
          foi criado com sucesso.
        </span>
      </div>
      <div className="row buttons">
        <button type="button" className="btn primary" onClick={handleFinish}>
          finalizar
        </button>
      </div>
    </Modal>
  );
};

Success.propTypes = {
  show: PropTypes.bool.isRequired,
  user: PropTypes.oneOfType([PropTypes.object]),
  onClose: PropTypes.func.isRequired,
};

Success.defaultProps = {
  user: {},
};

export default withRouter(Success);
