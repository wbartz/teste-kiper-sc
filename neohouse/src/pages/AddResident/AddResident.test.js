import { shallow } from 'enzyme';
import React from 'react';
import AddResident from './formAddResident';

describe('AddResident', () => {
  let wrapper;
  const props = {
    history: {
      push: jest.fn(),
    },
    location: {
      state: {
        apartment_id: 1,
      },
    },
    addResident: jest.fn(),
  };

  beforeEach(() => {
    wrapper = shallow(<AddResident {...props} />);
  });

  it('render without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
