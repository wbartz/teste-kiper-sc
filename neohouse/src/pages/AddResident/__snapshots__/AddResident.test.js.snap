// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddResident render without crashing 1`] = `ShallowWrapper {}`;
